### ISSUE TITLE
KeyError:'db' when capturing elasticsearch span context

### ISSUE THREAD

**Issue by brett-fitz** [elastic/apm-agent-python#1755]
**Describe the bug**: ...

**To Reproduce**

1. Instrument via capture serverless
2. Perform an async_scan:
```python
async with AsyncElasticsearch(**elastic_context) as client:
    async for doc in async_scan(
          client=client,
          index='XXX',
          query=query,
          _source=['field']
      ):
```
![image](https://user-images.githubusercontent.com/37186808/219222266-597464b9-9c70-4cea-b5de-f2c4b570cc5f.png)

**Environment (please complete the following information)**
- OS: [e.g. Linux] debian:bullseye-slim (python:3.9-slim container image) x86
- Python version: 3.9
- Framework and version [e.g. Django 2.1]: N/A AWS Lambda Function
- APM Server version: 8.6.1
- Agent version: 6.14.0

**Additional context**

- Agent config options 
  <details>
    <summary>Click to expand</summary>

    ```
    ELASTIC_APM_LOG_LEVEL: warning
    ELASTIC_APM_LAMBDA_APM_SERVER: https://<elastic-cloud-hostname>:443
    ELASTIC_APM_SECRET_TOKEN: ${env:ELASTIC_APM_SECRET_TOKEN}
    ELASTIC_APM_ENVIRONMENT: ${self:provider.stage}
    ELASTIC_APM_SPAN_COMPRESSION_SAME_KIND_MAX_DURATION: 50ms
    ```
  </details>
- `requirements.txt`:
  <details>
    <summary>Click to expand</summary>

    ```toml
    [tool.poetry.dependencies]
    python = "^3.9"
    elasticsearch = "^8"
    elastic-apm = "^6.14"
    tomlkit = "^0.11.1"
    uvloop = "^0.17.0"
    PyYAML = "^6.0"
    aiobotocore = "^2.3"
    urllib3 = "^1.26.10"
    python-dateutil = "^2.8.2"
    pyjarm = "^0.0.5"
    aiohttp = "^3.8"
    multidict = "^6"
    ```
  </details>

**Comment by basepi** [comment]
Thanks for the report. We fixed this in the synchronous version in #1690 and forgot to apply the fix to the async version. Apologies! Fix is here: #1758

**Comment by brett-fitz** [comment]
Awesome, thanks!
